src/main/java/com/mrcrayfish/controllable/client/ControllerInput.java

```diff
@@ -654,6 +671,110 @@ else if(dir < 0)
         }
     }
 
+    private List<Widget> getWidgets(Object parent)
+    {
+        return getWidgets(parent, 0);
+    }
+
+    private List<Widget> getWidgets(Object parent, int level) {
+        List<Widget> widgets = new ArrayList<>();
+        if (parent == null || (level != 0 && parent instanceof Screen)) return widgets;
+        Class clazz = parent.getClass();
+        if (clazz.isPrimitive()) return widgets;
+
+        if (parent instanceof Widget) {
+            widgets.add((Widget) parent);
+            return widgets;
+        }
+        List<Field> fields = ReflectionUtils.getFieldsIncludingSuperclasses(parent.getClass()).stream()
+                .filter(f -> !f.getType().isPrimitive()).collect(Collectors.toList());
+        List<Field> wFeilds = fields.stream()
+                .filter(f -> Widget.class.isAssignableFrom(f.getType()) || List.class.isAssignableFrom(f.getType()))
+                .collect(Collectors.toList());
+        try {
+            for (Field f : fields) {
+                boolean accessible = f.isAccessible();
+                f.setAccessible(true);
+                Object o = f.get(parent);
+                {
+                    if (o instanceof List) {
+                        List l = (List) f.get(parent);
+                        if (l != null && !l.isEmpty()) {
+                            widgets.addAll(
+                                    (Collection<? extends Widget>) l.stream()
+                                            .filter(i -> i instanceof Widget).collect(Collectors.toList())
+                            );
+                        }
+                    } else if (o instanceof Widget) {
+                        widgets.add((Widget) f.get(parent));
+                    }
+                }
+                f.setAccessible(accessible);
+            }
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+        }
+        if (level + 1 < 4) {
+            fields.removeAll(wFeilds);
+            fields.forEach(f -> {
+                boolean accessible = f.isAccessible();
+                f.setAccessible(true);
+                try {
+                    Object o = f.get(parent);
+                    if (o != null)
+                    {
+                        widgets.addAll(getWidgets(o, level + 1));
+                    }
+                } catch (IllegalAccessException e) {
+                    e.printStackTrace();
+                }
+                f.setAccessible(accessible);
+            });
```

Comment:
This is a huge detriment to performance. Who knows what classes this will search through and how many unrelated ones. Imagine a class that has 100 non-primative fields.